### PR TITLE
Feature/user dashboards

### PR DESCRIPTION
**Pull request by b-stein**
This PR contains the trip card creation for the DOM, and Traveler & Agent (class) methods for data manipulation finding activeTrips, pastTrips, upcomingTrips, money spent for year.

Initial test suites for both classes are setup as well. 2 domUpdates tests are present.

I'm happy with my generateTripCards() function in domUpdates. It's dynamic to intake any HTML element and data to insert. So with this I am only interpolating HTML elements in JS once, and calling on that function a few times for the 6 sections that need this functionality.

### CODE REVIEW

File: src/agent.js [hunk 1]
```diff
@@ -1,16 +1,59 @@
 import User from './user';
 
 class Agent extends User {
-	constructor(username, pwd) {
+	constructor(username, pwd, trips) {
 		super(username, pwd, true, true);
```
Comment (carflor) [inline src/agent.js:5]: Nice use of inheritance!

File: src/css/base.scss [hunk 2]
```diff
@@ -166,4 +173,62 @@ input {
...
+	transition: .8s ease;
+	width: 100%;
+}
+
+.trip-container {
+	display: flex;
+  flex-wrap: nowrap;
+	overflow-x: auto;
+	overflow-y: hidden;
+	min-height: 150px;
```
Comment (carflor) [inline src/css/base.scss:233]: a couple places where you might need to indent property styling. but the lint will probably catch these whenver you get to that! good job on styling.

File: test/domUpdates-test.js [hunk 3]
```diff
@@ -14,8 +14,8 @@ describe('domUpdates', () => {
 		expect(domUpdates).to.be.an('object');
 	})
 
-	it('should spy on ___', () => {
-		domUpdates.displayAgentDash();
-		expect(document.querySelector()).to.have.been.called(1);
+	it('should spy on querySelector functions being called on the document', () => {
+		domUpdates.querySelector();
+		expect(document.querySelector).to.have.been.called(1);
 	})
```
Comment (carflor) [inline test/domUpdates-test.js:20]: Nice!!

**Comment by b-stein** [comment]
@wvmitchell

**Pull request merged**